src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_debug.c

```diff
@@ -1,25 +1,1002 @@
+                        G_TX_RENDERTILE, 0, 0, 1 << 10, 1 << 10);
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+void KaleidoScope_DrawDebugEditor(PlayState* play) {
+    Gfx* gfx;
+    Gfx* gfxRef;
+    s32 counterDigits[4];
+    s16 slot;
+    s16 i;
+    s16 j;
+    s32 x;
+    s32 y;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    func_8012C628(play->state.gfxCtx);
+
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+    gDPSetCombineMode(POLY_OPA_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 0, 0, 0, 220);
+    gDPFillRectangle(POLY_OPA_DISP++, 24, 12, 298, 228);
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetCombineLERP(POLY_OPA_DISP++, 0, 0, 0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0, 0, 0, 0, PRIMITIVE, TEXEL0, 0,
+                      PRIMITIVE, 0);
+
+    gfxRef = POLY_OPA_DISP;
+    gfx = Graph_GfxPlusOne(gfxRef);
+    gSPDisplayList(OVERLAY_DISP++, gfx);
+
+    KaleidoScope_DrawDebugEditorText(&gfx);
+
+    gSPEndDisplayList(gfx++);
+    Graph_BranchDlist(gfxRef, gfx);
+    POLY_OPA_DISP = gfx;
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 0, 0, 255);
+    gDPSetEnvColor(POLY_OPA_DISP++, 0, 0, 0, 0);
+
+    // Current Health Quarter (X / 4)
+    KaleidoScope_DrawDigit(play, (((void)0, gSaveContext.save.playerData.health) % 16) / 4, 217, 15);
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+
+    // Rupees
+    counterDigits[0] = counterDigits[1] = counterDigits[2] = 0;
+    counterDigits[3] = gSaveContext.save.playerData.rupees;
+    while (counterDigits[3] >= 1000) {
+        counterDigits[0]++;
+        counterDigits[3] -= 1000;
+    }
+
+    while (counterDigits[3] >= 100) {
+        counterDigits[1]++;
+        counterDigits[3] -= 100;
+    }
+
+    while (counterDigits[3] >= 10) {
+        counterDigits[2]++;
+        counterDigits[3] -= 10;
+    }
+
+    for (i = 0, x = 68; i < 4; i++, x += 9) {
+        KaleidoScope_DrawDigit(play, counterDigits[i], x, 15);
+    }
+
+    // Health capacity
+    counterDigits[2] = 0;
+    counterDigits[3] = gSaveContext.save.playerData.healthCapacity / 0x10;
+    while (counterDigits[3] >= 10) {
+        counterDigits[2]++;
+        counterDigits[3] -= 10;
+    }
+
+    KaleidoScope_DrawDigit(play, counterDigits[2], 145, 15);
+    KaleidoScope_DrawDigit(play, counterDigits[3], 154, 15);
+
+    // Health
+    counterDigits[2] = 0;
+    counterDigits[3] = gSaveContext.save.playerData.health / 0x10;
+    while (counterDigits[3] >= 10) {
+        counterDigits[2]++;
+        counterDigits[3] -= 10;
+    }
+
+    KaleidoScope_DrawDigit(play, counterDigits[2], 195, 15);
+    KaleidoScope_DrawDigit(play, counterDigits[3], 204, 15);
+
+    // Inventory
+    for (slot = 0, i = 0, y = 38; i < 4; i++, y += 14) {
+        for (j = 0, x = 44; j < 6; j++, slot++, x += 23) {
```

Comment:
For smaller single focus functions using single letter iterators is fine, but I kinda want these iterators to have names when its something this complicated... except you can't name them for this one loop because they get reused. Could you include a comment explaining what each are (assuming they are known)

Reply:
They do kinda have meaning instead of generic variables, x, y, are the coordinates to print digits, while i, j are row/columns of the grid of numbers to print, but agree it does look like a garbled mess, so I can use better names

Reply:
It seems like `i` is always column and `j` is always row?

Reply:
It follows that pattern every time except for `items` and `masks` in which they're switched (and I tested swapping them and doesn't work).

And `j` is only used for 3 loops, and 2/3 (swapped) are for columns and 1/3 (normal) is for rows
And `i` is used for 11 loops, and 2/11 (swapped) are for rows, and 9/11 (normal) are for columns

Reply:
:despair: